### PR TITLE
feat: (platform) Refactor of SearchInput and renaming to SearchField

### PR DESCRIPTION
**Pull request by KevinOkamoto**
#### Please provide a link to the associated issue.

#1745 

#### Please provide a brief summary of this pull request.

This is a refactor of the SearchInput field based on the latest updates to Fundamental NGX and Fiori3. The following changes have been made to SearchInput:

* Changed the name of the component from SearchInput to SearchField in accordance with Fiori 3 naming convention.
* Removed dependency on core Dropdown component as it has been deprecated in the code.
* Added a "clear" text input button in accordance with Fiori 3 design.
* Added option to use DataSource similar to the DataSource used in the current platform ComboBox component.

Screen Shot 2020-03-25 at 4 57 01 PM

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

Documentation checklist:
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples

### CODE REVIEW

File: libs/platform/src/lib/components/search-field/search-field.component.html [hunk 1]
```diff
@@ -0,0 +1,77 @@
+<div class="search-input"
```
Comment (kavya-b) [inline libs/platform/src/lib/components/search-field/search-field.component.html]: is the class name in fiori3 still `search-input` or should this also be `search-field`?

Comment (KevinOkamoto) [reply libs/platform/src/lib/components/search-field/search-field.component.html]: It probable should be `search-field` for naming consistency. I'll make the change.

Comment (KevinOkamoto) [reply libs/platform/src/lib/components/search-field/search-field.component.html]: It has been changed.

File: libs/platform/src/lib/components/menu/menu.component.ts [hunk 2]
```diff
@@ -90,7 +90,7 @@ export class MenuComponent implements OnInit, AfterViewInit, AfterContentInit, O
      */
     @Output() close: EventEmitter<MenuCloseMethod> = new EventEmitter();
 
-    private _direction: 'ltr' | 'rtl' | void = 'ltr';
+    public direction: 'ltr' | 'rtl' | void = 'ltr';
```
Comment (salarenko) [inline libs/platform/src/lib/components/menu/menu.component.ts]: In what situation `void` is an expected type?

Comment (KevinOkamoto) [reply libs/platform/src/lib/components/menu/menu.component.ts]: This was left over from when a different service was used for directionality. I'll change it.

**Comment by InnaAtanasova** [comment]
Hi @KevinOkamoto , Search Field looks great! I have more a design/UX question regarding the PR. I noticed that the clear button is left outside of the focus outline. 
Screen Shot 2020-04-01 at 5 39 55 PM

This is causing one UX issue - when you start typing inside the field you can see this "jumping/moving" of the focus outline. I know from the UX experts that it's not good :) The other problem is that you visually separate the clear button from the input and I think it should be "inside" the outline. I found a couple of examples from the ui5 demo app:
Screen Shot 2020-04-01 at 5 36 47 PM

Screen Shot 2020-04-01 at 5 37 00 PM

Screen Shot 2020-04-01 at 5 36 32 PM

**Comment by InnaAtanasova** [comment]
Also, make sure @fkolar approves the PR before you merge. We agreed to have approvals from both teams before merging platform PRs.

**Comment by fkolar** [comment]
@KevinOkamoto I just see one change in the future but we are not ready yet and the work is in progress. This the thing we talked about for the Compact (contentdensity), disabled, and all these properties so inputsearch will inhert from BaseComponent with all these common behavior.

**Pull request merged**